install.sh

```diff
@@ -0,0 +1,635 @@
+
+To get started with AgentStack, run:
+    exec \$SHELL
+    agentstack init
+
+For more information, run:
+    agentstack docs -or- agentstack quickstart
+EOF
+)
+
+MSG_ALREADY_INSTALLED=$(cat <<EOF
+✅ AgentStack is already installed.
+Run 'agentstack update' to update to the latest version.
+EOF
+)
+
+MSG_UNINSTALL=$(cat <<EOF
+✅ AgentStack has been uninstalled.
+If you encountered any issues, or have feedback, please open an issue:
+    $REPO_URL/issues
+EOF
+)
+
+usage() {
+    cat <<EOF
+agentstack-install.sh
+
+The installer for AgentStack
+
+This script installs uv the Python package manager, installs a compatible Python 
+version ($PYTHON_VERSION), and installs AgentStack.
+
+USAGE:
+    agentstack-install.sh [OPTIONS]
+
+OPTIONS:
+    uninstall                  Uninstall
+    --version=<version>        Specify version to install (default: $VERSION)
+    --python-version=<version> Specify Python version to install (default: $PYTHON_VERSION)
+    --dev-branch=<branch>      Install from a specific git branch/commit/tag
+    --verbose                  Enable verbose output
+    --quiet                    Suppress output
+    -h, --help                 Show this help message
+EOF
+}
+# TODO allow user to specify install path with --target
+# TODO allow user to specify Python version with --python-version
+# TODO uninstall
+
+say() {
+    if [ "1" = "$PRINT_QUIET" ]; then
+        echo -e "$1"
+    fi
+}
+
+say_verbose() {
+    if [ "1" = "$PRINT_VERBOSE" ]; then
+        echo -e "[DEBUG] $1"
+    fi
+}
+
+ACTIVITY_PID=""
+_show_activity() {
+    while true; do
+        echo -n "."
+        sleep 1
+    done
+}
+
+show_activity() {
+    if [ "1" = "$PRINT_QUIET" ] || [ "1" = "$PRINT_VERBOSE" ]; then
+        return 0
+    fi
+    _show_activity &
+    ACTIVITY_PID=$!
+    # trap end_activity EXIT
+    # trap 'kill $ACTIVITY_PID' INT
+    # wait $ACTIVITY_PID
+}
+
+end_activity() {
+    if [ -n "$ACTIVITY_PID" ]; then
+        say ""  # newline after the dots
+        kill $ACTIVITY_PID
+    fi
+}
+
+err() {
+    end_activity
+    if [ "1" = "$PRINT_QUIET" ]; then
+        local _red=$(tput setaf 1 2>/dev/null || echo '')
+        local _reset=$(tput sgr0 2>/dev/null || echo '')
+        say "\n${_red}[ERROR]${_reset}: $1" >&2
+        say "\nRun with --verbose for more details."
+        say "\nIf you need help, please feel free to open an issue:"
+        say "  $REPO_URL/issues\n"
+    fi
+    exit 1
+}
+
+err_missing_cmd() {
+    local _cmd_name=$1
+    local _help_text=""
+    local _platform=$(platform)
+
+    if [ $_platform == "linux" ]; then
+        if [ $_cmd_name == "gcc" ]; then
+            _help_text="Hint: sudo apt-get install build-essential"
+        else
+            _help_text="Hint: sudo apt-get install $_cmd_name"
+        fi
+    elif [ $_platform == "macos" ]; then
+        _help_text="Hint: brew install $_cmd_name"
+    fi
+    err "A required dependency is missing. Please install: $*\n$_help_text"
+}
```

Comment:
i see why gcc might be a blocker. def something to keep an eye on

Reply:
couple notes based on my research into this:

we can't handle the apt install for them because it needs sudo. although i've seen script prompt for a sudo password before. 

we can install `brew` in userspace on macos and then get `gcc`. it comes with apple's dev tools, but that's a big package that not everyone has. 

we can also install `linuxbrew` in userspace and then get `gcc` on linux/WSL.

Reply:
i recently installed bun with their script and it stopped and told me to install a dependency then try again, so i feel like this is fine